variables.tf

```diff
@@ -57,31 +59,37 @@ variable "dns_project" {
 }
 
 variable "dns_ttl" {
+  type        = number
   description = "The DNS TTL in seconds for records created in Cloud DNS.  The default value should be used unless the application demands special handling."
-  default     = "300"
+  default     = 300
 }
 
 variable "dns_managed_zone" {
+  type        = string
   description = "The name of the managed zone to create records within.  This managed zone must exist in the host project."
   default     = ""
 }
 
```

Comment:
missed type for enable_cloud_dns, and why it's string, not bool?

Reply:
I can rework logic, if we need it, but:
  dns_args_missing         = var.enable_cloud_dns != "" && var.dns_domain == "" && var.dns_project == "" && var.dns_managed_zone == "" ? 1 : 0
  ptr_args_missing         = var.enable_reverse_dns != "" && var.dns_reverse_zone == "" ? 1 : 0
  dns_fqdns                = formatlist("%s.%s", var.dns_short_names, var.dns_domain)
  regional_addresses_count = var.global ? 0 : length(var.names)
  global_addresses_count   = var.global ? length(var.names) : 0
  dns_forward_record_count = var.enable_cloud_dns != "" ? length(local.dns_fqdns) : 0
  dns_reverse_record_count = var.enable_reverse_dns != "" ? length(local.dns_fqdns) : 0

Reply:
Yes it's ugly, I've reworked it and testing new approach now, Thank you for review!